### ISSUE TITLE
Add option to restrain the selection of genome assemblies with submission date before or after a date

### ISSUE THREAD

**Issue by dawnmy** [pirovc/genome_updater#42]
Thank you for developing `genome_updater` which makes genome downloading very convenient. Is it possible to add an option to restrain the selection of genome assemblies with submission date before or after a given date. For example a `--date` option:
`--date <20190227` will select genome before this day while `--date >20190227` will select the genomes after this day.

**Comment by pirovc** [comment]
Hi @dawnmy 

You can use the custom filter option (`-F`) to set specific dates you want. For example `-F "15:2019/10/29,2019/10/30"`. Unfortunatelly date ranges are not yet supported. For now, if you need longer periods, I'd suggest to download the complete assembly summary for refseq or genbank, filter the summary and use it as an input file to genome_updater. Something like:

```sh
wget https://ftp.ncbi.nlm.nih.gov/genomes/refseq/assembly_summary_refseq.txt # OR https://ftp.ncbi.nlm.nih.gov/genomes/genbank/assembly_summary_genbank.txt

# for "--date >20190227"
awk -v dstart="20190227" -v dend="" -F "\t" '{gsub("/","",$15); if(($15>=dstart || dstart=="") && ($15<=dend || dend=="")) print $15}' assembly_summary_refseq.txt > filtered_assembly_summary_refseq.txt
# for "--date <20190227"
awk -v dstart="" -v dend="20190227" -F "\t" '{gsub("/","",$15); if(($15>=dstart || dstart=="") && ($15<=dend || dend=="")) print $15}' assembly_summary_refseq.txt > filtered_assembly_summary_refseq.txt

genome_updater.sh -e filtered_assembly_summary_refseq.txt # ... plus all other parameters you may need
```

I'll mark this issue as an enhancement and try to add it in future releases.

**Comment by dawnmy** [comment]
Thank you for the solution. And looking forward to the new release

**Comment by dawnmy** [comment]
@pirovc The selection of assemblies by organism groups `-g` does not work for external assembly_summary files. I tried to run:
```
genome_updater.sh -g "viral" -d "refseq"  -l "Complete Genome" -f "genomic.fna.gz" \
 -o "viral_refseq_taxid_top3" -t 10 -m -A 3 -a -p -e filtered_assembly_summary_refseq.txt
```
with version 3.0, but no only viral genomes were downloaded.

**Comment by pirovc** [comment]
Indeed, it won't work in this case. You could either re-download the specific viral assembly summary (https://ftp.ncbi.nlm.nih.gov/genomes/refseq/viral/assembly_summary.txt) and do the date filtration again or run the current command with `-T 10239` to select only [viral](https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?mode=Info&id=10239&lvl=3&lin=f&keep=1&srchmode=1&unlock) assemblies.
